### DISCUSSION TITLE
During damage simulation,why prompt "dataset "phi" not found"

### DISCUSSION THREAD

**Discussion by tutfighting** [damask-multiphysics/DAMASK#14] [Q&A] [answered]
Material.yaml:
---
homogenization:
  SX:
    N_constituents: 1
    mechanical:
      type: pass
    damage:
      type: pass
      initialDamage: 1.0
      output: [phi]
      
phase:
  '1':
    lattice: cF
    mechanical:
      output: [F, P, F_e, F_p, L_p, O]
      elastic: {type: Hooke, C_11: 107e9, C_12: 52e9, C_44: 26e9}
      plastic:
        type: dislotwin
        output: [rho_dip, rho_mob]
        D: 1.525e-05
        tau_0: [55e6]
        N_sl: [12]
        b_sl: [2.546e-10]
        rho_dip_0: [1]
        rho_mob_0: [1e12]
        v_0: [10e-5]
        Q_sl: [8.361e-20]
        Q_cl: 1.876e-19
        p_sl: [0.8]
        q_sl: [1.5]
        i_sl: [45]
        V_at: 6.6e-29
        D_a: 7
        D_0: 0.000623
        h_sl-sl: [1, 1, 1, 1, 1, 1, 1]
        
    damage:
        type: anisobrittle
        
        output: [f_phi]

        N_cl: [3]

        g_crit: [0.5e+7]
        s_crit: [0.006666]
        dot_o_0: 1.e-3
        p: 20

        l_c: 1.0
        mu: 0.001
    
material:
  - homogenization: SX
    constituents:
      - phase: 1
        v: 1.0
        O: [1.0, 0.0, 0.0, 0.0]
  - homogenization: SX
    constituents:
      - phase: 1
        v: 1.0
        O: [0.7936696712125002, -0.28765777461664166, -0.3436487135089419, 0.4113964260949434]
  - homogenization: SX
    constituents:
      - phase: 1
        v: 1.0
        O: [0.3986143167493579, -0.7014883552495493, 0.2154871765709027, 0.5500781677772945]
  - homogenization: SX
    constituents:
      - phase: 1
        v: 1.0
        O: [0.28645844315788244, -0.022571491243423537, -0.467933059311115, -0.8357456192708106]
  - homogenization: SX
    constituents:
      - phase: 1
        v: 1.0
        O: [0.33012772942625784, -0.6781865350268957, 0.6494525351030648, 0.09638521992649676]
  - homogenization: SX
    constituents:
      - phase: 1
        v: 1.0
        O: [0.43596817439583935, -0.5982537129781701, 0.046599032277502436, 0.6707106499919265]
  - homogenization: SX
    constituents:
      - phase: 1
        v: 1.0
        O: [0.169734823419553, -0.699615227367322, -0.6059581215838098, -0.33844257746495854]
  - homogenization: SX
    constituents:
      - phase: 1
        v: 1.0
        O: [0.9698864809294915, 0.1729052643205874, -0.15948307917616958, 0.06315956884687175]
  - homogenization: SX
    constituents:
      - phase: 1
        v: 1.0
        O: [0.46205660912967883, 0.3105054068891252, -0.617849551030653, 0.555294529545738]
  - homogenization: SX
    constituents:
      - phase: 1
        v: 1.0
        O: [0.4512443497461787, -0.7636045534540555, -0.04739348426715133, -0.45939142396805815]
  - homogenization: SX
    constituents:
      - phase: 1
        v: 1.0
        O: [0.2161856212656443, -0.6581450184826598, -0.5498086209601588, 0.4667112513346289]
  - homogenization: SX
    constituents:
      - phase: 1
        v: 1.0
        O: [0.8753220715350803, -0.4561599367657419, -0.13298279533852678, -0.08969369719975541]
  - homogenization: SX
    constituents:
      - phase: 1
        v: 1.0
        O: [0.11908260752431069, 0.18266024809834172, -0.7144822594012615, -0.664807992845101]
  - homogenization: SX
    constituents:
      - phase: 1
        v: 1.0
        O: [0.751104669484278, 0.5585633382623958, -0.34579336397009175, 0.06538900566860861]
  - homogenization: SX
    constituents:
      - phase: 1
        v: 1.0
        O: [0.08740438971703973, 0.8991264096610437, -0.4156704205935976, 0.10559485570696363]
  - homogenization: SX
    constituents:
      - phase: 1
        v: 1.0
        O: [0.5584325870096193, 0.6016408353068798, -0.14280340445801173, 0.5529814994483859]
  - homogenization: SX
    constituents:
      - phase: 1
        v: 1.0
        O: [0.4052725440888093, 0.25253073423599154, 0.5693263597910454, -0.669215876471182]
  - homogenization: SX
    constituents:
      - phase: 1
        v: 1.0
        O: [0.7570164606888676, 0.15265448024694664, -0.5998021466848317, 0.20942796551297105]
  - homogenization: SX
    constituents:
      - phase: 1
        v: 1.0
        O: [0.6987659297138081, -0.132172211261028, -0.19693254724422338, 0.6748883269678543]
  - homogenization: SX
    constituents:
      - phase: 1
        v: 1.0
        O: [0.7729330445886478, 0.21682179052722322, -0.5207379472917645, 0.2905078484066341]

tensionX.yaml:
---
solver:
    mechanical: spectral_basic
    damage: spectral

loadstep:
  - boundary_conditions:
      mechanical:
        dot_F: [[1.0e-3, 0, 0],
                [0,      x, 0],
                [0,      0, x]]
        P: [[x, x, x],
            [x, 0, x],
            [x, x, 0]]
    discretization:
      t: 10
      N: 40
    f_out: 4
  - boundary_conditions:
      mechanical:
        dot_F: [[1.0e-3, 0, 0],
                [0,      x, 0],
                [0,      0, x]]
        P: [[x, x, x],
            [x, 0, x],
            [x, x, 0]]
    discretization:
      t: 60
      N: 60
    f_out: 4

 and the .vti is the  20grains16x16x16.vti in the example grid files

**Comment by hyharry** [comment, marked as answer]
You probably forgot to specify the initial condition in your `vti` file. This can be simply done by the following snippet

```python
g = damask.GeomGrid.load(fn_vti)
g.initial_conditions['phi'] = 1
g.save(fn_vti)
```

with this updated `fn_vti` file, you can start your damage simulation.

**Reply by tutfighting** [reply]
I really appreciate your help！Now I have encountered a new problem，in my results in paraview,  [f_phi ]  output not be found, and  [phi] output no value changed, can you help me to solve this problem? or give me a example? Thank you again for your help ！！！

**Reply by hyharry** [reply]
if you are using the latest release or master branch of DAMASK, the output label name is `Psi_D`, then you can view the damage internal state in paraview. The unchanged damage field `phi` probably means that your microstructure does not have damage yet. 

I advice you to read through [the related paper](https://www.sciencedirect.com/science/article/abs/pii/S0022509616304513), then you will understand more your simulations and results.

**Reply by TranMHoang** [reply]
Hi, can I ask what the 's_crit' parameter refer to, since it is not present in the given paper or the DAMASK paper?

**Comment by SunW317** [comment]
Minimum|Maximum|Delta Damage      = 1.000000  1.000000   0.0000E+00

 ===========================================================================

 ┌─────────────────────────────────────────────────────────────────────┐
 │                        error                                        │
 │                        950                                          │
 ├─────────────────────────────────────────────────────────────────────┤
 │ max number of cut back exceeded, terminating                        │
 │                                                                     │
 └─────────────────────────────────────────────────────────────────────┘
According to your description, I'm running DAMASK, but it's giving me an error, what's the problem?

**Reply by hyharry** [reply]
This means that your simulation **does not converge**. **The reason can be quite complex**, which can be damage model you choose, parameters you choose (both mech and damage parameters), timestep is too coarse, etc. Remember the current `isobrittle` and `anisobrittle` damage model is essentially `brittle`, meaning damage can happen very fast, to resolve it, one should use a very tiny timestep at the onset of damage. 

Regarding the error you got, it means that it exceeds the default cutback number. To set it you can have a look at the [numerics.yaml](https://github.com/damask-multiphysics/DAMASK/blob/master/examples/config/numerics.yaml) and increase `N_cutback_max`. When you do it and put it in your running folder please run damask as `DAMASK_grid --geom g.vti --load l.yaml --material m.yaml --numerics your_new_numerics.yaml`

**Reply by SunW317** [reply]
Thank you so much for providing a response, it helps a lot! Thank you.

**Reply by SunW317** [reply]
Can both DAMASK-alpha7 and beta versions simulate damage and applied heat sources? I found that the file on the official website for alpha7 has less numerics.yaml than beta, does this mean that the alpha7 version is not perfect enough, and can the alpha7 version simulate damage and applied heat sources properly?

**Reply by hyharry** [reply]
The functionality has not changed from alpha7 to beta for your use case. But as you already noticed, the numerical option labels have changed, and you should always use the corresponding ones, if you want to change them.

**Reply by SunW317** [reply]
I understand what you mean, but I tried to change the value of (N_cutback_max) in the (numerics.yaml) file and during the simulation I found that the maximum cutback value was not changed.In the same case I changed the number of iterations (itmax) and the number of iterations did change during the simulation, so I didn't find a reason.Is my keyword wrong?Do you have a good suggestion?

**Reply by hyharry** [reply]
As I said, those option keys have changed. According to the alpha7 version you mentioned, you should look up for the keys corresponding to your version, in case you cannot find it, you can check your directory and file `damask/example/config/numerics.yaml`. 

By the way `itmax` is another numerics option which is different from the cut back option.
